testing/src/test/java/org/motechproject/nms/testing/it/kilkari/InboxServiceBundleIT.java

```diff
@@ -0,0 +1,217 @@
+import org.ops4j.pax.exam.ExamFactory;
+import org.ops4j.pax.exam.junit.PaxExam;
+import org.ops4j.pax.exam.spi.reactors.ExamReactorStrategy;
+import org.ops4j.pax.exam.spi.reactors.PerSuite;
+
+@RunWith(PaxExam.class)
+@ExamReactorStrategy(PerSuite.class)
+@ExamFactory(MotechNativeTestContainerFactory.class)
+public class InboxServiceBundleIT extends BasePaxIT {
+
+	@Inject
+	private SubscriberService subscriberService;
+	@Inject
+	private SubscriptionService subscriptionService;
+	@Inject
+	private InboxService inboxService;
+	@Inject
+	private SubscriberDataService subscriberDataService;
+	@Inject
+	private SubscriptionPackDataService subscriptionPackDataService;
+	@Inject
+	private SubscriptionPackMessageDataService subscriptionPackMessageDataService;
+	@Inject
+	private SubscriptionDataService subscriptionDataService;
+	@Inject
+	private LanguageDataService languageDataService;
+	@Inject
+	private InboxCallDetailRecordDataService inboxCallDetailRecordDataService;
+	@Inject
+	private InboxCallDataDataService inboxCallDataDataService;
+	@Inject
+	private LanguageLocationDataService languageLocationDataService;
+	@Inject
+	private StateDataService stateDataService;
+	@Inject
+	private DistrictDataService districtDataService;
+	@Inject
+	private CircleDataService circleDataService;
+
+	private LanguageLocation gLanguageLocation;
+	private SubscriptionPack pregnancyPack;
+
+	private void createLanguageAndSubscriptionPacks() {
+		District district = new District();
+		district.setName("District 1");
+		district.setRegionalName("District 1");
+		district.setCode(1L);
+
+		District district2 = new District();
+		district2.setName("District 2");
+		district2.setRegionalName("District 2");
+		district2.setCode(2L);
+
+		State state = new State();
+		state.setName("State 1");
+		state.setCode(1L);
+		state.getDistricts().add(district);
+		state.getDistricts().add(district2);
+
+		stateDataService.create(state);
+
+		Language language = new Language("tamil");
+		languageDataService.create(language);
+
+		LanguageLocation languageLocation = new LanguageLocation("10",
+				new Circle("AA"), language, true);
+		languageLocation.getDistrictSet().add(district);
+		languageLocationDataService.create(languageLocation);
+
+		language = new Language("english");
+		languageDataService.create(language);
+
+		languageLocation = new LanguageLocation("99", new Circle("BB"),
+				language, true);
+		languageLocation.getDistrictSet().add(district2);
+		gLanguageLocation = languageLocationDataService
+				.create(languageLocation);
+
+		pregnancyPack = subscriptionPackDataService.create(
+				SubscriptionPackBuilder.createSubscriptionPack(
+						"pregnancyPack",
+						SubscriptionPackType.PREGNANCY, 			
+						SubscriptionPackBuilder.PREGNANCY_PACK_WEEKS,	//72
+						2));											// messages per week
+
+		pregnancyPack = subscriptionPackDataService.byName("pregnancyPack"); 
+	}
+
+	private void cleanupData() {
```

Comment:
When creating new code (or updating existing code) please use TestingService.clearDatabase() to delete the database. That will reduce code duplication & potential errors.